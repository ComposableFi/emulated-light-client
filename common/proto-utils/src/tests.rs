pub mod pb {
    pub mod foo {
        #[derive(Clone, PartialEq, prost::Message)]
        pub struct Message {
            #[prost(uint64, tag = "1")]
            pub value: u64,
        }

        impl prost::Name for Message {
            const NAME: &'static str = "Message";
            const PACKAGE: &'static str = "foo";

            fn full_name() -> ::alloc::string::String { "foo.Message".into() }

            fn type_url() -> ::alloc::string::String { "/foo.Message".into() }
        }
    }
}

crate::define_message! {
    pub use pb::foo::Message;
    test_message Self { value: 42 };
}

#[test]
fn test_macro() {
    // Check that test_foo test has been defined.
    #[allow(dead_code)]
    if false {
        test_message();
    }
    let msg: Message = Message::test();
    assert_eq!(42, msg.value);
}
