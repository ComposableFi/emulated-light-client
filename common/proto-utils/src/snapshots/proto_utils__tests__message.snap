---
source: common/proto-utils/src/tests.rs
expression: any.value
---
[
    8,
    42,
]
